OrbitGl/GpuTrack.cpp

```diff
@@ -103,15 +137,28 @@ float GpuTrack::GetYFromTimer(const TimerInfo& timer_info) const {
   if (collapse_toggle_->IsCollapsed()) {
     adjusted_depth = 0.f;
   }
-  return pos_[1] - layout.GetTextBoxHeight() * (adjusted_depth + 1.f) -
-         adjusted_depth * layout.GetSpaceBetweenGpuDepths();
+  if (timer_info.type() == TimerInfo::kGpuDebugMarker) {
+    return pos_[1] - layout.GetTextBoxHeight() * (adjusted_depth + 1.f);
+  }
+  CHECK(timer_info.type() == TimerInfo::kGpuActivity ||
+        timer_info.type() == TimerInfo::kGpuCommandBuffer);
+
+  if (has_vulkan_layer_command_buffer_timers_) {
+    adjusted_depth *= 2.f;
+  }
+
+  float gap_space = adjusted_depth * layout.GetSpaceBetweenGpuDepths();
+  if (timer_info.type() == TimerInfo::kGpuCommandBuffer) {
+    ++adjusted_depth;
+  }
+  return pos_[1] - layout.GetTextBoxHeight() * (adjusted_depth + 1.f) - gap_space;
 }
 
-// When track is collapsed, only draw "hardware execution" timers.
+// When track is collapsed, only draw "hardware execution" timers and "debug markers".
 bool GpuTrack::TimerFilter(const TimerInfo& timer_info) const {
   if (collapse_toggle_->IsCollapsed()) {
     std::string gpu_stage = string_manager_->Get(timer_info.user_data_key()).value_or("");
-    if (gpu_stage != kHwExecutionString) {
+    if (gpu_stage != kHwExecutionString && timer_info.type() != TimerInfo::kGpuDebugMarker) {
```

Comment:
I think this is more readable:

`return gpu_stage == kHwExecutionString || timer_info.type() == TimerInfo::kGpuDebugMarker;`

Reply:
Yes, thanks!